apps/server/src/apps/helpers/app-logger-middleware.ts

```diff
@@ -0,0 +1,28 @@
+import { Request, Response, NextFunction } from 'express';
+import { Logger } from '@src/core/logger';
+import { RequestLoggable } from '@src/apps/helpers/request-loggable';
+import { Configuration } from '@hpi-schul-cloud/commons/lib';
+
+export const createAppLoggerMiddleware = (
+	logger: Logger
+): ((request: Request, response: Response, next: NextFunction) => void) => {
+	logger.setContext('AppLoggerMiddleware');
+	const enabled = Configuration.get('REQUEST_LOGGING_ENABLED') as boolean;
+
+	return (request: Request, response: Response, next: NextFunction): void => {
+		if (enabled) {
+			const startAt = process.hrtime();
+			const { method, originalUrl } = request;
+
+			response.on('finish', () => {
```

Comment:
Reagiert response.on('finish' wenn der Timeout Interceptor den request unterbricht? Könnte man das noch in ein Test ablegen?